esp/src/eclwatch/FileBloomsWidget.js

```diff
@@ -0,0 +1,122 @@
+            if (currSel && !currSel.initalized) {
+                if (currSel.id === this.id + "_Grid") {
+                    this.refreshGrid()
+                } else if (currSel.id === this.fileBloomsWidget.id && !this.fileBloomsWidget.initalized) {
+                    this.fileBloomsWidget.init({
+                        firstLoad: true
+                    });
+                } else {
+                    currSel.init(currSel.params);
+                }
+            }
+        },
+
+        postCreate: function (args) {
+            var context = this;
+            this.inherited(arguments);
+            this.openButton = registry.byId(this.id + "Open");
+            this.refreshButton = registry.byId(this.id + "Refresh");
+        },
+
+        createGrid: function (domID) {
+            var context = this;
+            var retVal = new declare([ESPUtil.Grid(true, true)])({
+                store: this.store,
+                columns: {
+                    col1: selector({
+                        width: 10,
+                        selectorType: "checkbox",
+                        disabled: function (item) {
+                            if (item.type === "item") {
+                                return false;
+                            }
+                            return true;
+                        },
+                        sortable: false,
+                        unhidable: true
+                    }),
+                    Name:{
+                        width: 200,
+                        collapseOnRefresh: false,
+                        label: nlsHPCC.FieldNames,
+                        sortable: false
+                    },
+                    LimitName: { label: this.i18n.Limit, width: 100, sortable: false },
+                    ProbabilityName: { label: this.i18n.Probability, width: 100, sortable: false }
+                }
+            }, domID);
+
+            retVal.on(".dgrid-row:dblclick", function (evt) {
+                evt.preventDefault();
+                context.grid.refresh()
+            });
+
+            return retVal;
+        },
+
+        _onRefresh: function () {
+            this.refreshGrid();
+        },
+
+        refreshGrid: function () {
+            var context = this;
+            WsDfu.DFUInfo({
+                request: {
+                    Name: context.params.Name
+                }
+            }).then(function (response) {
+                var results = [];
+
+                if (lang.exists("DFUInfoResponse.FileDetail.Blooms.DFUFileBloom", response)) {
+                    arrayUtil.forEach(response.DFUInfoResponse.FileDetail.Blooms.DFUFileBloom, function (row, idx) {
+                        lang.mixin(row, {
+                            Name: row.FieldNames.Item,
+                            ProbabilityName: row.Probability,
+                            LimitName: row.Limit,
+                            type: "fieldname"
```

Comment:
Do you still need this type? Usually add these for renderCell methods to perform an action on those rows.